packages/core/src/bodl/bodl-emitter-service.ts

```diff
@@ -0,0 +1,187 @@
+import { LineItemMap } from '../cart';
+import { Checkout, CheckoutService } from '../checkout';
+import { ShopperCurrency } from '../config';
+import { Order } from '../order';
+
+import BodlService from "./bodl-service";
+import { BodlEventsCheckout } from './bodl-window';
+
+export default class BodlEmitterService implements BodlService {
+    private _checkoutStarted = false;
+
+    constructor(
+        private checkoutService: CheckoutService,
+        private bodlEvents: BodlEventsCheckout
+    ) {
+
+    }
+
+    checkoutBegin(): void {
+        if (this._checkoutStarted) {
+            return;
+        }
+
+        const checkout = this.getCheckout();
+
+        if (!checkout) {
+            return;
+        }
+
+
+        const {
+            cart: {
+                cartAmount,
+                currency,
+                lineItems,
+                id,
+                coupons
+            },
+        } = checkout;
+
+        this.bodlEvents.emit('create_checkout_begin', {
+            id,
+            currency: currency.code,
+            cart_value: cartAmount,
+            coupon: (coupons || []).map(coupon => coupon.code.toUpperCase()).join(','),
```

Comment:
Do we need to do the `||` here? Since if there are no coupons applied it's value is `[]`

Reply:
makes sense